src/poetry/core/packages/dependency.py

```diff
@@ -279,10 +279,13 @@ def is_directory(self) -> bool:
     def is_url(self) -> bool:
         return False
 
-    def to_pep_508(self, with_extras: bool = True) -> str:
+    def to_pep_508(self, with_extras: bool = True, resolved: bool = False) -> str:
         from poetry.core.packages.utils.utils import convert_markers
 
-        requirement = self.base_pep_508_name
+        if resolved and hasattr(self, "base_pep_508_name_resolved"):
+            requirement: str = self.base_pep_508_name_resolved
+        else:
+            requirement = self.base_pep_508_name
```

Comment:
I didn't write any new tests for this line, but since the default value of "resolved" is False this means that any invocation of to_pep_508 without the resolved keyword will hit the else branch of this if statement. In particular all the already existing tests for to_pep_508 in  `test_dependency.py` and `test_vcs_dependency.py` will cover this line.

Reply:
You're right. Only the condition `resolved==True` and `hasattr(self, "base_pep_508_name_resolved") == False` was not tested.